File: Back/src/main/java/com/MediaWeseco/Back/service/AuthService.java
package com.MediaWeseco.Back.service;

import com.MediaWeseco.Back.dtos.AuthResponse;
import com.MediaWeseco.Back.dtos.GoogleCompleteDto;
import com.MediaWeseco.Back.dtos.LoginDto;
import com.MediaWeseco.Back.dtos.RegisterInitDto;
import com.MediaWeseco.Back.dtos.SetPasswordDto;
import com.MediaWeseco.Back.dtos.VerifyCodeDto;

public interface AuthService {

    // Paso 1: Registro Inicial (Guarda datos y genera código)
    String registerInit(RegisterInitDto dto);

    // Paso 2: Verificar el código del correo
    void verifyCode(VerifyCodeDto dto);

    // Paso 3: Crear contraseña final
    void setPassword(SetPasswordDto dto);

    // Paso Extra: Completar perfil de Google
    AuthResponse completeGoogleProfile(GoogleCompleteDto dto);

    // Login: Devuelve el Token JWT (String)
    AuthResponse login(LoginDto dto);

    // Devuelve el JWT si existe, o lanza una excepción especial si es nuevo
    AuthResponse loginWithGoogle(String googleToken);
}

File: Front/src/services/AuthService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/auth';

export default {
    async login(identifier, password) {
        const response = await axios.post(`${API_URL}/login`, { identifier, password });
        return response.data;
    },

    async loginWithGoogle(googleToken) {
        const response = await axios.post(`${API_URL}/google`, { token: googleToken });
        return response.data;
    },

    // --- REGISTRO (NUEVO) ---

    // Paso 1: Enviar datos iniciales
    async registerInit(user) {
        // user debe tener: email, username, fechaNacimiento
        const response = await axios.post(`${API_URL}/register`, user);
        return response.data;
    },

    // Paso 2: Enviar código de verificación
    async verifyCode(email, code) {
        const response = await axios.post(`${API_URL}/verify`, {
            email,
            verificationCode: code
        });
        return response.data;
    },

    // Paso 3: Crear contraseña
    async setPassword(email, password) {
        const response = await axios.post(`${API_URL}/set-password`, {
            email,
            password
        });
        return response.data;
    }
};